CODESTYLE.md

```diff
@@ -0,0 +1,208 @@
+# Kowala Style Guide
+
+This guide covers Kowala programming style from aesthetic issues to conventions and coding standard.
+
+## Test
+
+All PR should be accompanied by unit tests
+
+### Tools
+
+### Unit tests
+
+We use 
+
+[Testify]([https://github.com/stretchr/testify) "A toolkit with common assertions and mocks that plays nicely with the standard library"
+
+[Mockery](https://github.com/vektra/mockery) "A mock code autogenerator for golang"
+
+Mocks should go into a subpackage of the interface that implements
+to generate a Mock for a interface run:
+
+```go
+mockery -name InterfaceName
+```
+
+### Test fixtures
+
+We use Golden Files trick, to keep our test fixtures up to date
```

Comment:
It there something to link to in order to explain snapshot testing? is it worth using a snapshot library? I build a [really small one](https://github.com/endiangroup/snaptest).